presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -146,12 +148,15 @@ public static long fromUnixTime(@SqlType(StandardTypes.DOUBLE) double unixTime)
     @SqlType(StandardTypes.TIMESTAMP_WITH_TIME_ZONE)
     public static long fromUnixTime(@SqlType(StandardTypes.DOUBLE) double unixTime, @SqlType(StandardTypes.BIGINT) long hoursOffset, @SqlType(StandardTypes.BIGINT) long minutesOffset)
     {
+        int offset = toIntExact(hoursOffset * 60 + minutesOffset);
```

Comment:
This could be under the `try` too. This way you will handle overflow in `toIntExact` too.

Reply:
Actually we can't use `toIntExact` here, because it seems like the offset can be negative and that will change the current behavior (e.g., `select from_unixtime(1, 9223372036854775807,9223372036854775807);`, works before, will fail with this change).

Reply:
`from_unixtime(1, 9223372036854775807,9223372036854775807)` works... but I doubt it works _correctly_.